### PR TITLE
Adding Associations

### PR DESCRIPTION
**Pull request by SnakyMz**
Things I have done:

- [ ] Created class Classroom
- [ ] Added relationship between Classroom and Student classes
- [ ] Created class Book
- [ ] Created class Rental
- [ ] Added relationship between Book and Rental classes
- [ ] Added relationship between Person and Rental classes

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -2,9 +2,12 @@
...
+  attr_accessor :classroom
+
   def initialize(classroom, age, name = 'Unknown', parent_permission: true)
-    super(name, age, parent_permission)
+    super(age, name, parent_permission)
     @classroom = classroom
+    classroom.add_student(self)
   end
 
   def play_hooky
```
Comment (Johnadibe) [inline student.rb]: > - [ ] Good job @SnakyMz, In accordance with the project's requirements which can be found [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#:~:text=Make%20sure%20that%20when%20setting%20the%20classroom%20for%20a%20student%20it%20also%20adds%20it%20to%20the%20classrooms%27%20students.), when setting the classroom for a student it also adds it to the classrooms' students. I recommend creating a method to achieve this. This enhancement will be highly beneficial.👍
> 
> You can use the below as a guide:
> 
>  ```
> def classroom=(classroom)
>       @classroom = classroom
>       classroom.students.push(self) unless classroom.students.include?(self)
>   end
> 
> ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative 'rental'
+
+# Class containing data about books
+class Book
+  attr_accessor :title, :author
```
Comment (Johnadibe) [inline book.rb]: > - [ ] It appears that you haven't included the getter and setter attributes when creating the has-many relationship for Book and Rental (a book has many rentals). Kindly implement them. 😊
> ```suggestion
>   attr_accessor :title, :author, :rental
> ```

File: person.rb [hunk 3]
```diff
@@ -1,16 +1,22 @@
 require_relative 'nameable'
+require_relative 'rental'
 
 # Class containing data about person
 class Person < Nameable
   attr_accessor :name, :age
```
Comment (Johnadibe) [inline person.rb]: > - [ ] It appears that you haven't included the getter and setter attributes when creating the has-many side of Person and Rental (a person has many rentals). Kindly implement them. 😊
> ```suggestion
>   attr_accessor :name, :age, :rental
> ```

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @SnakyMz👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ Created a class Classroom✔️
✅ Created the has-many/belongs-to relationship between Classroom and Student✔
✅ Created a class Book✔️
✅ Created a class Rental✔
✅ Created the many-to-many relationship between Person and Book✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 2 above)
Comment (SnakyMz) [reply book.rb]: Thank you for the suggestion

File: person.rb (hunk 3 above)
Comment (SnakyMz) [reply person.rb]: Thank you for the suggestion

File: student.rb (hunk 1 above)
Comment (SnakyMz) [reply student.rb]: Thanks for the suggestion, I will make changes as suggested

**Review by uwadonat — APPROVED** [review]
## Hi @SnakyMz,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Good commit messages ✔️
- Well-organized and clean project ✔️
- Detailed pull request message ✔️

approval1


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**